### PR TITLE
Feature/added op consultation text field in op triage/consultation form

### PR DESCRIPTION
**Pull request by Pai026**
Added OP Consultation in the drop menu of Decision after OP Triage/Consultation and displays a free text space when the same is selected..

Issue #381

### CODE REVIEW

**Comment by tomahawk-pilot** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151&bid=19014353)

File: src/Components/Facility/ConsultationForm.tsx [hunk 1]
```diff
@@ -30,28 +56,31 @@ const initForm: any = {
   referred_to: "",
   examination_details: "",
   existing_medication: "",
-  prescribed_medication: ""
+  prescribed_medication: "",
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:59]: Codacy found an issue: [Identifier 'prescribed_medication' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 2]
```diff
@@ -187,11 +235,17 @@ export const ConsultationForm = (props: any) => {
       setIsLoading(true);
       const data = {
         symptoms: state.form.symptoms,
-        other_symptoms: state.form.otherSymptom ? state.form.other_symptoms : undefined,
-        symptoms_onset_date: state.form.hasSymptom ? state.form.symptoms_onset_date : undefined,
+        other_symptoms: state.form.otherSymptom
+          ? state.form.other_symptoms
+          : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:240]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 3]
```diff
@@ -163,10 +205,16 @@ export const ConsultationForm = (props: any) => {
           }
           return;
         case "referred_to":
-          if (state.form.suggestion === 'R' && !state.form[field]) {
+          if (state.form.suggestion === "R" && !state.form[field]) {
             errors[field] = "Please select the referred to facility";
             invalidForm = true;
           }
+        case "OPconsultation":
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:212]: Codacy found an issue: [Expected a 'break' statement before 'case'.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 4]
```diff
@@ -187,11 +235,17 @@ export const ConsultationForm = (props: any) => {
       setIsLoading(true);
       const data = {
         symptoms: state.form.symptoms,
-        other_symptoms: state.form.otherSymptom ? state.form.other_symptoms : undefined,
-        symptoms_onset_date: state.form.hasSymptom ? state.form.symptoms_onset_date : undefined,
+        other_symptoms: state.form.otherSymptom
+          ? state.form.other_symptoms
+          : undefined,
+        symptoms_onset_date: state.form.hasSymptom
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:241]: Codacy found an issue: [Identifier 'symptoms_onset_date' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 5]
```diff
@@ -200,19 +254,26 @@ export const ConsultationForm = (props: any) => {
         discharge_date: state.form.discharge_date,
         patient: patientId,
         facility: facilityId,
-        referred_to: state.form.suggestion === 'R' ? state.form.referred_to : undefined,
+        referred_to:
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:257]: Codacy found an issue: [Identifier 'referred_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 6]
```diff
@@ -101,11 +137,17 @@ export const ConsultationForm = (props: any) => {
...
+            hasSymptom:
+              !!res.data.symptoms &&
+              !!res.data.symptoms.length &&
+              !!res.data.symptoms.filter((i: number) => i !== 1).length,
+            otherSymptom:
+              !!res.data.symptoms &&
+              !!res.data.symptoms.length &&
+              !!res.data.symptoms.filter((i: number) => i === 9).length,
+            admitted: res.data.admitted ? String(res.data.admitted) : "false",
+            admitted_to: res.data.admitted_to ? res.data.admitted_to : "",
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:149]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 7]
```diff
@@ -200,19 +254,26 @@ export const ConsultationForm = (props: any) => {
         discharge_date: state.form.discharge_date,
         patient: patientId,
         facility: facilityId,
-        referred_to: state.form.suggestion === 'R' ? state.form.referred_to : undefined,
+        referred_to:
+          state.form.suggestion === "R" ? state.form.referred_to : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:258]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 8]
```diff
@@ -187,11 +235,17 @@ export const ConsultationForm = (props: any) => {
       setIsLoading(true);
       const data = {
         symptoms: state.form.symptoms,
-        other_symptoms: state.form.otherSymptom ? state.form.other_symptoms : undefined,
-        symptoms_onset_date: state.form.hasSymptom ? state.form.symptoms_onset_date : undefined,
+        other_symptoms: state.form.otherSymptom
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:238]: Codacy found an issue: [Identifier 'other_symptoms' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 9]
```diff
@@ -163,10 +205,16 @@ export const ConsultationForm = (props: any) => {
           }
           return;
         case "referred_to":
-          if (state.form.suggestion === 'R' && !state.form[field]) {
+          if (state.form.suggestion === "R" && !state.form[field]) {
             errors[field] = "Please select the referred to facility";
             invalidForm = true;
           }
+        case "OPconsultation":
+          if (state.form.suggestion === "OP" && !state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:213]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 10]
```diff
@@ -187,11 +235,17 @@ export const ConsultationForm = (props: any) => {
...
+        other_symptoms: state.form.otherSymptom
+          ? state.form.other_symptoms
+          : undefined,
+        symptoms_onset_date: state.form.hasSymptom
+          ? state.form.symptoms_onset_date
+          : undefined,
         suggestion: state.form.suggestion,
         admitted: JSON.parse(state.form.admitted),
-        admitted_to: JSON.parse(state.form.admitted) ? state.form.admitted_to : undefined,
+        admitted_to: JSON.parse(state.form.admitted)
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:246]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 11]
```diff
@@ -163,10 +205,16 @@ export const ConsultationForm = (props: any) => {
...
           return;
         case "referred_to":
-          if (state.form.suggestion === 'R' && !state.form[field]) {
+          if (state.form.suggestion === "R" && !state.form[field]) {
             errors[field] = "Please select the referred to facility";
             invalidForm = true;
           }
+        case "OPconsultation":
+          if (state.form.suggestion === "OP" && !state.form[field]) {
+            errors[field] = "Please enter OP consultation Details";
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:214]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 12]
```diff
@@ -163,10 +205,16 @@ export const ConsultationForm = (props: any) => {
           }
           return;
         case "referred_to":
-          if (state.form.suggestion === 'R' && !state.form[field]) {
+          if (state.form.suggestion === "R" && !state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:208]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

**Comment by tomahawk-pilot** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151&bid=19014353)

File: src/Components/Facility/ConsultationForm.tsx [hunk 13]
```diff
@@ -163,10 +205,17 @@ export const ConsultationForm = (props: any) => {
...
             errors[field] = "Please select the referred to facility";
             invalidForm = true;
           }
+          return;
+        case "OPconsultation":
+          if (state.form.suggestion === "OP" && !state.form[field]) {
+            errors[field] = "Please enter OP consultation Details";
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:215]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 14]
```diff
@@ -163,10 +205,17 @@ export const ConsultationForm = (props: any) => {
           }
           return;
         case "referred_to":
-          if (state.form.suggestion === 'R' && !state.form[field]) {
+          if (state.form.suggestion === "R" && !state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:208]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 15]
```diff
@@ -163,10 +205,17 @@ export const ConsultationForm = (props: any) => {
...
             errors[field] = "Please select the referred to facility";
             invalidForm = true;
           }
+          return;
+        case "OPconsultation":
+          if (state.form.suggestion === "OP" && !state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:214]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 1 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:59]: Codacy found an issue: [Identifier 'prescribed_medication' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 6 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:149]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 16]
```diff
@@ -187,11 +236,17 @@ export const ConsultationForm = (props: any) => {
       setIsLoading(true);
       const data = {
         symptoms: state.form.symptoms,
-        other_symptoms: state.form.otherSymptom ? state.form.other_symptoms : undefined,
-        symptoms_onset_date: state.form.hasSymptom ? state.form.symptoms_onset_date : undefined,
+        other_symptoms: state.form.otherSymptom
+          ? state.form.other_symptoms
+          : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:241]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 17]
```diff
@@ -200,19 +255,26 @@ export const ConsultationForm = (props: any) => {
         discharge_date: state.form.discharge_date,
         patient: patientId,
         facility: facilityId,
-        referred_to: state.form.suggestion === 'R' ? state.form.referred_to : undefined,
+        referred_to:
+          state.form.suggestion === "R" ? state.form.referred_to : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:259]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 18]
```diff
@@ -200,19 +255,26 @@ export const ConsultationForm = (props: any) => {
         discharge_date: state.form.discharge_date,
         patient: patientId,
         facility: facilityId,
-        referred_to: state.form.suggestion === 'R' ? state.form.referred_to : undefined,
+        referred_to:
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:258]: Codacy found an issue: [Identifier 'referred_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 19]
```diff
@@ -187,11 +236,17 @@ export const ConsultationForm = (props: any) => {
...
+        other_symptoms: state.form.otherSymptom
+          ? state.form.other_symptoms
+          : undefined,
+        symptoms_onset_date: state.form.hasSymptom
+          ? state.form.symptoms_onset_date
+          : undefined,
         suggestion: state.form.suggestion,
         admitted: JSON.parse(state.form.admitted),
-        admitted_to: JSON.parse(state.form.admitted) ? state.form.admitted_to : undefined,
+        admitted_to: JSON.parse(state.form.admitted)
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:247]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 20]
```diff
@@ -187,11 +236,17 @@ export const ConsultationForm = (props: any) => {
       setIsLoading(true);
       const data = {
         symptoms: state.form.symptoms,
-        other_symptoms: state.form.otherSymptom ? state.form.other_symptoms : undefined,
-        symptoms_onset_date: state.form.hasSymptom ? state.form.symptoms_onset_date : undefined,
+        other_symptoms: state.form.otherSymptom
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:239]: Codacy found an issue: [Identifier 'other_symptoms' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx [hunk 21]
```diff
@@ -187,11 +236,17 @@ export const ConsultationForm = (props: any) => {
...
+          ? state.form.other_symptoms
+          : undefined,
+        symptoms_onset_date: state.form.hasSymptom
```
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:242]: Codacy found an issue: [Identifier 'symptoms_onset_date' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

**Comment by dauntlessnomad** [comment]
@Pai026 Fix the merge conflicts and reference the issue for the pr

**Comment by dauntlessnomad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151&bid=19014353)

File: src/Components/Facility/ConsultationForm.tsx (hunk 14 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:208]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 18 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:258]: Codacy found an issue: [Identifier 'referred_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 20 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:239]: Codacy found an issue: [Identifier 'other_symptoms' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 13 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:215]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 19 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:247]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 21 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:242]: Codacy found an issue: [Identifier 'symptoms_onset_date' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 6 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:149]: Codacy found an issue: [Identifier 'admitted_to' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 17 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:259]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 16 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:241]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 1 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:59]: Codacy found an issue: [Identifier 'prescribed_medication' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

File: src/Components/Facility/ConsultationForm.tsx (hunk 15 above)
Comment (tomahawk-pilot) [inline src/Components/Facility/ConsultationForm.tsx:214]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5808151)

**Pull request merged**